zigpy/zcl/clusters/general.py

```diff
@@ -1302,6 +1310,449 @@ class PollControl(Cluster):
+        ep.in_clusters[zigpy.zcl.clusters.general.Basic.cluster_id]._update_attribute(
+            0x0004, "GreenPower"
+        )
+        in_clusters = None
+        out_clusters = None
+        if type is not None and type in GreenPowerProxy.device:
+            name, in_clusters, out_clusters = GreenPowerProxy.device[type]
+            ep.in_clusters[
+                zigpy.zcl.clusters.general.Basic.cluster_id
+            ]._update_attribute(0x0005, name)
+            for cluster_id in in_clusters:
+                LOGGER.debug("Add input cluster id %s on device %s", cluster_id, ieee)
+                ep.add_input_cluster(cluster_id)
+            for cluster_id in out_clusters:
+                LOGGER.debug("Add output cluster id %s on device %s", cluster_id, ieee)
+                ep.add_output_cluster(cluster_id)
+        else:
+            ep.in_clusters[
+                zigpy.zcl.clusters.general.Basic.cluster_id
+            ]._update_attribute(0x0005, "GreenPowerDevice")
+        ep = dev.add_endpoint(self.endpoint_id)
+        ep.status = zigpy.endpoint.Status.ZDO_INIT
+        ep.profile_id = zigpy.profiles.zha.PROFILE_ID
+        ep.device_type = zigpy.profiles.zha.DeviceType.GREEN_POWER
+        ep.add_input_cluster(self.cluster_id)
+        application.device_initialized(dev)
+        return dev
+
+    def handle_notification(
+        self, ieee, header, counter, command_id, payload, payload_length, mic
+    ):
+        application = self.endpoint.device.application
+        if ieee not in application.devices:
+            return
+        if command_id not in GreenPowerProxy.command:
+            LOGGER.debug("Unhandled command_id : %s", command_id)
+            return
+        calcul_mic = self.calcul_mic(
+            ieee,
+            header,
+            counter,
+            command_id.to_bytes(1, "little")
+            + payload.to_bytes(payload_length, "little"),
+            payload_length + 1,
+        )
+        if calcul_mic is not None and calcul_mic != mic:
+            LOGGER.debug(
+                "Wrong mic : %s, calcul mic %s, ignore frame", hex(mic), hex(calcul_mic)
+            )
+            return
+        (
+            command,
+            type,
+            schema,
+            cluster_id,
+            zcl_command_id,
+            value,
+        ) = GreenPowerProxy.command[command_id]
+        payload, _ = t.deserialize(payload.to_bytes(payload_length, "little"), schema)
+        LOGGER.debug(
+            "Green power frame ieee : %s, command_id : %s, payload : %s,counter : %s",
+            ieee,
+            command_id,
+            payload,
+            counter,
+        )
+        value = value + tuple(payload)
+        dev = application.devices[ieee]
+        if counter is not None:
+            attributes = (
+                dev.endpoints[self.endpoint_id].in_clusters[self.cluster_id]._attr_cache
+            )
+            if 0x9999 in attributes and attributes[0x9999] > counter:
```

Comment:
Wouldn't you want `>=` instead of `>`? Otherwise, while this would prevent the counter from decrementing, it would allow the same frame to be received multiple times.

Reply:
Well seen. Thank you it's fixed